src/main/java/com/beerjournal/breweriana/collection/persistence/UserCollectionRepository.java

```diff
@@ -85,15 +109,21 @@ int updateItem(Item item) {
                         Criteria.where("itemRefs").elemMatch(Criteria.where("itemId").is(item.getId()))
                 )),
                 new Update().set("itemRefs.$.name", item.getName())
-                            .set("itemRefs.$.type", item.getType()),
+                        .set("itemRefs.$.type", item.getType()),
                 UserCollection.class);
 
         return writeResult.getN();
     }
 
-    private List<Item> findAllNotIn(Set<String> userItemsNames) {
-        return mongoOperations.find(
-                new Query(Criteria.where("name").not().in(userItemsNames)),
-                Item.class);
+    private Page<Item> findAllNotIn(Set<String> userItemsNames, PageRequest pageRequest, String filterVariableName, String filterVariableValue) {
+        Criteria criteria = Criteria.where("name").not().in(userItemsNames);
+        if (!Strings.isNullOrEmpty(filterVariableValue))
+            criteria = criteria.andOperator(Criteria.where(filterVariableName).regex(Pattern.compile("^" + filterVariableValue)));
+
+        Query query = new Query(criteria);
+        long total = mongoOperations.count(query, Item.class);
+        List<Item> items = mongoOperations.find(query.with(pageRequest), Item.class);
```

Comment:
Problem with pagination with query (http://stackoverflow.com/questions/29030542/pagination-with-mongotemplate).

So I workaround it with http://stackoverflow.com/questions/20366666/spring-data-mongo-pagination (phuong answear.)